pkg/operator/controller/ingress_status.go

```diff
@@ -249,33 +140,75 @@ func (c *ingressStatusCache) computeLoadBalancerStatus(ic *operatorv1.IngressCon
 	conditions = append(conditions, operatorv1.OperatorCondition{
 		Type:    operatorv1.LoadBalancerManagedIngressConditionType,
 		Status:  operatorv1.ConditionTrue,
-		Reason:  "HasLoadBalancerEndpointPublishingStrategy",
-		Message: "IngressController has LoadBalancer endpoint publishing strategy",
+		Reason:  "WantedByEndpointPublishingStrategy",
+		Message: "The endpoint publishing strategy supports a managed load balancer",
 	})
 
 	switch {
-	case c.contains(&corev1.ServiceList{}, "pending-for", ic.Name):
+	case service == nil:
 		conditions = append(conditions, operatorv1.OperatorCondition{
 			Type:    operatorv1.LoadBalancerReadyIngressConditionType,
 			Status:  operatorv1.ConditionFalse,
-			Reason:  "LoadBalancerPending",
-			Message: "The LoadBalancer service is pending",
+			Reason:  "LoadBalancerNotFound",
+			Message: "The LoadBalancer service resource is missing",
 		})
-	case c.contains(&corev1.ServiceList{}, "provisioned-for", ic.Name):
+	case isProvisioned(service):
 		conditions = append(conditions, operatorv1.OperatorCondition{
 			Type:    operatorv1.LoadBalancerReadyIngressConditionType,
 			Status:  operatorv1.ConditionTrue,
 			Reason:  "LoadBalancerProvisioned",
 			Message: "The LoadBalancer service is provisioned",
 		})
-	default:
+	case isPending(service):
+		reason := "LoadBalancerPending"
+		message := "The LoadBalancer service is pending"
+
+		// Try and find a more specific reason for for the pending status.
+		createFailedReason := "CreatingLoadBalancerFailed"
+		failedLoadBalancerEvents := getEventsByReason(operandEvents, "service-controller", createFailedReason)
```

Comment:
Perhaps these should be sorted by time, descending?

Reply:
This would be useful once we check for more than 1 event type, but otherwise it is not strictly necessary since we ignore events once the LB is provisioned.